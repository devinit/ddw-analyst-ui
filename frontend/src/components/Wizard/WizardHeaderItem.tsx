import React, { FunctionComponent } from 'react';
import { Nav, NavItemProps } from 'react-bootstrap';

const WizardHeaderItem: FunctionComponent<NavItemProps> = ({ children, ...props }) => {
  return <Nav.Item {...props}>{children}</Nav.Item>;
};

export { WizardHeaderItem };
