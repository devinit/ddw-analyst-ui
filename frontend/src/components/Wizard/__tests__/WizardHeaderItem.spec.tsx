/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { WizardHeaderItem } from '../WizardHeaderItem';

describe('WizardHeaderItem', () => {
  test('renders the default structure correctly', () => {
    const renderer = TestRenderer.create(<WizardHeaderItem />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with children', () => {
    const renderer = TestRenderer.create(
      <WizardHeaderItem>
        <div>My Child</div>
      </WizardHeaderItem>,
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
