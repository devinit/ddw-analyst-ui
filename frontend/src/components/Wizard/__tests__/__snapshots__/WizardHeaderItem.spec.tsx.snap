// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardHeaderItem renders the default structure correctly 1`] = `
<div
  className="nav-item"
/>
`;

exports[`WizardHeaderItem renders with children 1`] = `
<div
  className="nav-item"
>
  <div>
    My Child
  </div>
</div>
`;
